inference/sources/fly/main_test.go

```diff
@@ -0,0 +1,249 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	. "github.com/vlab-research/vlab/inference/inference-data"
+	. "github.com/vlab-research/vlab/inference/test-helpers"
+)
+
+func check(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+func resData(fi string) string {
+	b, e := ioutil.ReadFile(fmt.Sprintf("test/%s", fi))
+	check(e)
+
+	return string(b)
+}
+
+func dataAssertions(t *testing.T, e []*InferenceDataEvent) {
+	assert.Equal(t, 8, len(e))
+
+	// assert.Equal(t, "21085286190ffad1248d17c4135ee56f", e[0].User.ID)
+	// assert.Equal(t, "5fcb3f9c162e1fcdaadff4405b741080", e[10].User.ID)
+	// assert.Equal(t, "3btcnj9rrhzyttmghhapu6znz3y43i36", e[19].User.ID)
+
+	// assert.Equal(t, "126", e[0].User.ID)
+	// assert.Equal(t, "127", e[1].User.ID)
+	// assert.Equal(t, "128", e[2].User.ID)
+
+}
+
+func TestGetResponses_PaginatesWhenPageIsFull(t *testing.T) {
+	res1 := resData("fly_example.json")
+	res2 := resData("fly_example2.json")
+
+	s := string(res1)
+	data := GetResponsesResponse{}
+	json.Unmarshal([]byte(s), &data)
+
+	s2 := string(res2)
+	data2 := GetResponsesResponse{}
+	json.Unmarshal([]byte(s2), &data2)
+
+	// lastToken := data.Items[len(data.Items)-1]
+
+	count := 0
+	ts, _ := TestServer(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/all", r.URL.Path)
+		after := r.URL.Query().Get("after")
+
+		if count == 0 {
+			assert.Condition(t, func() bool {
+				if after == "" {
+					return after != ""
+				}
+				return true
+			}, "No token sent")
+
+			assert.Equal(t, "oldtoken", after)
+			w.WriteHeader(200)
+			w.Header().Set("Content-Type", "application/json")
+			fmt.Fprint(w, res1)
+		}
+
+		if count == 1 {
+			assert.Equal(t, "4viu4r8djwxwb2udbivx42avnawwj5wj", after)
+			w.WriteHeader(200)
+			w.Header().Set("Content-Type", "application/json")
+			fmt.Fprint(w, res2)
+		}
+		count++
+
+	})
+
+	tc := flyConnector{BaseUrl: ts.URL, Key: "sosecret", PageSize: 3}
+
+	events := tc.GetResponses(&Source{StudyID: res1}, "formfoo", "oldtoken", 0)
+
+	e := Sliceit(events)
+	dataAssertions(t, e)
```

Comment:
Add an assertion for `count == 2` and some assertions on the data and this looks good!